streams/src/test/java/org/apache/kafka/streams/integration/KStreamsFineGrainedAutoResetIntegrationTest.java

```diff
@@ -83,7 +83,7 @@ public static void startKafkaCluster() throws Exception {
     public void setUp() throws Exception {
 
         Properties props = new Properties();
-        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "none");
```

Comment:
Can you add a new test with `none` and `@Test(expected = StreamsException.class)`. Additionally, catch the exception and verify that cause is `NoOffsetForPartitionException` before re-throwing original exception -- if cause does not match, fail the test.